### PR TITLE
Add preset device trust roles

### PR DESCRIPTION
**Pull request by codingllama**
Add the following preset/example roles:

* device-admin (as shown by [device trust docs][device-docs])
* device-enroll (lets users enroll their devices)
* require-trusted-device (meant as an example/starting point for enforcement)

[device-docs]: https://goteleport.com/docs/access-controls/device-trust/guide/?scope=enterprise#step-12-register-a-trusted-device

### CODE REVIEW

**Comment by codingllama** [comment]
As created in the server:

```yaml
kind: role
metadata:
  description: Administer trusted devices
  id: 1692725645911361000
  labels:
    teleport.internal/resource-type: preset
  name: device-admin
spec:
  allow:
    rules:
    - resources:
      - device
      verbs:
      - list
      - create
      - read
      - update
      - delete
      - create_enroll_token
      - enroll
  deny: {}
  options:
    cert_format: standard
    create_db_user: false
    create_desktop_user: false
    desktop_clipboard: true
    desktop_directory_sharing: true
    enhanced_recording:
    - command
    - network
    forward_agent: false
    idp:
      saml:
        enabled: true
    max_session_ttl: 30h0m0s
    pin_source_ip: false
    port_forwarding: true
    record_session:
      default: best_effort
      desktop: true
    ssh_file_copy: true
version: v6
```

```yaml
kind: role
metadata:
  description: Enroll trusted devices
  id: 1692367644874884000
  name: device-enroll
spec:
  allow:
    rules:
    - resources:
      - device
      verbs:
      - enroll
  deny: {}
  options:
    cert_format: standard
    create_db_user: false
    create_desktop_user: false
    desktop_clipboard: true
    desktop_directory_sharing: true
    enhanced_recording:
    - command
    - network
    forward_agent: false
    idp:
      saml:
        enabled: true
    max_session_ttl: 30h0m0s
    pin_source_ip: false
    port_forwarding: true
    record_session:
      default: best_effort
      desktop: true
    ssh_file_copy: true
version: v6
```

```yaml
kind: role
metadata:
  description: Require trusted device to access resources
  id: 1692725645922416000
  labels:
    teleport.internal/resource-type: preset
  name: require-trusted-device
spec:
  allow:
    db_labels:
      '*': '*'
    db_names:
    - '*'
    db_users:
    - '*'
    kubernetes_groups:
    - '{{internal.kubernetes_groups}}'
    - system:masters
    - developers
    - viewers
    kubernetes_labels:
      '*': '*'
    logins:
    - '{{internal.logins}}'
    node_labels:
      '*': '*'
  deny: {}
  options:
    cert_format: standard
    create_db_user: false
    create_desktop_user: false
    desktop_clipboard: true
    desktop_directory_sharing: true
    device_trust_mode: required
    enhanced_recording:
    - command
    - network
    forward_agent: false
    idp:
      saml:
        enabled: true
    max_session_ttl: 30h0m0s
    pin_source_ip: false
    port_forwarding: true
    record_session:
      default: best_effort
      desktop: true
    ssh_file_copy: true
version: v6
```

**Review by zmb3 — APPROVED** [review]

**Comment by codingllama** [comment]
FYI @sshahcodes.

**Comment by codingllama** [comment]
Tests should be good now, sorry for the delay.

**Review by jakule — APPROVED** [review]

File: lib/services/presets.go [hunk 1]
```diff
@@ -364,6 +364,118 @@ func NewPresetGroupAccessRole() types.Role {
...
+				Logins: []string{"{{internal.logins}}"},
+				NodeLabels: types.Labels{
+					types.Wildcard: []string{types.Wildcard},
+				},
+
+				// All k8s nodes.
+				KubeGroups: []string{
+					"{{internal.kubernetes_groups}}",
+					// Common/example groups.
+					"system:masters",
```
Comment (sshahcodes) [inline lib/services/presets.go:460]: `system:masters` is this necessary? We can skip the reference to super privilege otherwise.

Comment (codingllama) [reply lib/services/presets.go:460]: Not necessary, but we do need to match groups and this is an often used / powerful one. The entire role is more of an example, to be modified as people want.

Comment (codingllama) [reply lib/services/presets.go:460]: Adding a bit more, I don't think you need to call this out explicitly in docs, if that's what you meant.

Comment (codingllama) [reply lib/services/presets.go:460]: No worries. It was mailed and merged fairly quickly.

**Pull request merged**